src/mlpack/core/util/split_data.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef __MLPACK_CORE_UTIL_SPLIT_DATA_HPP
+#define __MLPACK_CORE_UTIL_SPLIT_DATA_HPP
+
+#include <mlpack/core.hpp>
+
+#include <iterator>
+#include <numeric>
+#include <random>
+#include <tuple>
+#include <vector>
+
+namespace mlpack {
+namespace util {
+
+/**
+ *Split training data and test data
+ *@param input input data want to split
+ *@param label input label want to split
+ *@param trainData training data split by input
+ *@param testData test data split by input
+ *@param trainLabel train label split by input
+ *@param testLabel test label split by input
+ *@param testRatio the ratio of test data
+ *@code
+ *arma::mat input = loadData();
+ *arma::Row<size_t> label = loadLabel();
+ *arma::mat trainData;
+ *arma::mat testData;
+ *arma::Row<size_t> trainLabel;
+ *arma::Row<size_t> testLabel;
+ *arma::arma_rng::set_seed(100); //set the seed if you like
+ *TrainTestSplit(input, label, trainData,
+ *               testData, trainLabel, testLabel);
+ *@endcode
+ */
+template<typename T, typename U>
+void TrainTestSplit(arma::Mat<T> const &input,
+                    arma::Row<U> const &inputLabel,
+                    arma::Mat<T> &trainData,
+                    arma::Mat<T> &testData,
+                    arma::Row<U> &trainLabel,
+                    arma::Row<U> &testLabel,
+                    double const testRatio)
+{
+  size_t const testSize =
+      static_cast<size_t>(input.n_cols * testRatio);
```

Comment:
Because the compiler may give you type cast warning if you do not add static_cast